nemo/collections/asr/data/data_simulation.py

```diff
@@ -0,0 +1,1324 @@
+        ):
+            raise Exception("Speaker enforcement end is outside of [0,1]")
+
+        if (
+            self._params.data_simulator.session_params.min_dominance
+            * self._params.data_simulator.session_config.num_speakers
+            > 1
+        ):
+            raise Exception("Number of speakers times minimum dominance is greater than 1")
+
+        if (
+            self._params.data_simulator.session_params.overlap_prob
+            / self._params.data_simulator.session_params.turn_prob
+            > 1
+        ):
+            raise Exception("Overlap probability / turn probability is greater than 1")
+        if (
+            self._params.data_simulator.session_params.overlap_prob
+            / self._params.data_simulator.session_params.turn_prob
+            == 1
+            and self._params.data_simulator.session_params.mean_silence > 0
+        ):
+            raise Exception("Overlap probability / turn probability is equal to 1 and mean silence is greater than 0")
+
+        if (
+            self._params.data_simulator.session_params.window_type not in ['hamming', 'hann', 'cosine']
+            and self._params.data_simulator.session_params.window_type != None
+        ):
+            raise Exception("Incorrect window type provided")
+
+    def _get_speaker_ids(self):
+        """
+        Randomly select speaker IDs from loaded manifest file
+
+        Returns:
+            speaker_ids (list): Speaker IDs
+        """
+        speaker_ids = []
+        s = 0
+        while s < self._params.data_simulator.session_config.num_speakers:
+            file = self._manifest[np.random.randint(0, len(self._manifest) - 1)]
+            # fn = file['audio_filepath'].split('/')[-1]
+            # speaker_id = fn.split('-')[0]
+            speaker_id = file['speaker_id']
+            if speaker_id not in speaker_ids:  # ensure speaker ids are not duplicated
+                speaker_ids.append(speaker_id)
+                s += 1
+        return speaker_ids
+
+    def _get_speaker_samples(self, speaker_ids):
+        """
+        Get a list of the samples for each of the specified speakers
+
+        Args:
+            speaker_ids (list): LibriSpeech speaker IDs for each speaker in the current session.
+        Returns:
+            speaker_lists (dict): Dictionary of manifest lines per speaker
+        """
+        speaker_lists = {}
+        for i in range(0, self._params.data_simulator.session_config.num_speakers):
+            speaker_lists[str(speaker_ids[i])] = []
+        # loop over manifest and add files corresponding to each speaker to each sublist
+        for file in self._manifest:
+            # fn = file['audio_filepath'].split('/')[-1]
+            # new_speaker_id = fn.split('-')[0]
+            new_speaker_id = file['speaker_id']
+            for spid in speaker_ids:
+                if spid == new_speaker_id:
+                    speaker_lists[str(spid)].append(file)
```

Comment:
you may not require for loop here, since you already created empty dictionary with speaker ids as keys, you can check for speaker  in speaker ids with O(1) and append to speaker_list[spid]